### ISSUE TITLE
MyPy Unpack fails to resolve TypedDict imported from an upstream repo

### ISSUE THREAD

**Issue by LarsDu** [python/mypy#16710]
**Bug Report**

When importing a TypedDict from an upstream repository, MyPy fails to resolve that this object is a TypedDict. Only TypedDict subclassing objects defined in the same project are resolved correctly.

**To Reproduce**

```python
# In an upstream repo called `some_upstream_repo`

from typing import TypedDict

from typing_extensions import NotRequired

class TransformParams(TypedDict, total=False):
    param1: int
    param2: NotRequired[str]

-------
# In a downstream project that imports `some_upstream_repo`
from some_upstream_repo.transform_params import TransformParams
from typing_extensions import Unpack


def foo(**kwargs: Unpack[TransformParams]) -> None:
    print("This raises linter error because TransformParams is in an upstream repo")


class SomeParams(TypedDict, total=False):
    param1: NotRequired[int]


def bar(**kwargs: Unpack[SomeParams]) -> None:
    print("This works totally fine")

```

**Expected Behavior**
mypy outputs:
```
There should be no linting errors whatsover
```

**Actual Behavior**
```
note: In function "foo":
error: Unpack item in ** argument must be a TypedDict  [misc]
```

The imported TransformParams object is a TypedDict, but since its in an imported repo, it does not get resolved as a TypedDict. Only TypedDict subclassing objects _defined in the current repository_ resolve correctly to TypedDicts!

**Your Environment**
python 

- Mypy version used: 1.8.0
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files):
In pyproject.toml
```
[tool.mypy]
pretty = true
enable_incomplete_feature = "Unpack"
color_output = true

show_column_numbers = true
show_error_context = true
show_error_codes = true

ignore_missing_imports = true
follow_imports = "skip"

# Warnings
warn_redundant_casts = true
warn_unused_ignores = true
warn_unreachable = true

# Additional Checks
check_untyped_defs = true
no_implicit_optional = true
strict_optional = true

# Make sure we are using MyPy
disallow_untyped_calls = true
disallow_untyped_defs = true
```

- Python version used: 3.10.13

**Comment by hauntsaninja** [comment]
```
ignore_missing_imports = true
follow_imports = "skip"
```
You're telling mypy to not actually follow imports, so it doesn't know that the thing is a TypedDict 

`follow_imports = "skip"` is incredibly evil and I don't know why people keep using it. As the docs say, do not use it unless you know exactly what it does.

`ignore_missing_imports = true` is also not great and will hide misconfigurations of mypy. Try using `--disable-error-code=import-untyped` instead.

**Comment by LarsDu** [comment]
Ahh thank you. There were a number of setting here that were set over the course of development which I do not believe we fully understood.

**Comment by LarsDu** [comment]
Setting things to `follow_imports="normal"` and `disable_error_code = ["import-untyped"]` still leads to the same warning

**Comment by hauntsaninja** [comment]
You'll need to add a py.typed file to the upstream repo for mypy to be able to resolve it
